v2/googlecloud-to-neo4j/src/main/java/com/google/cloud/teleport/v2/neo4j/utils/DataCastingUtils.java

```diff
@@ -0,0 +1,350 @@
+          Date parsedDate = sdf.parse(row.getValue(fieldName));
+          ZonedDateTime zdt = ZonedDateTime.from(parsedDate.toInstant());
+          map.put(fieldName, zdt);
+        } else if (type.getTypeName().isDateType()) {
+          ReadableDateTime dt = row.getDateTime(fieldName);
+          ZonedDateTime zdt =
+              ZonedDateTime.ofLocal(
+                  LocalDateTime.of(
+                      dt.getYear(),
+                      dt.getMonthOfYear(),
+                      dt.getDayOfMonth(),
+                      dt.getHourOfDay(),
+                      dt.getMinuteOfHour(),
+                      dt.getSecondOfMinute(),
+                      dt.getMillisOfSecond() * 1_000_000),
+                  ZoneId.of(dt.getZone().getID(), ZoneId.SHORT_IDS),
+                  ZoneOffset.ofTotalSeconds(dt.getZone().getOffset(dt) / 1000));
+          map.put(fieldName, zdt);
+        } else if (type.typesEqual(Schema.FieldType.BOOLEAN)) {
+          map.put(fieldName, Boolean.parseBoolean(row.getBoolean(fieldName) + ""));
+        } else {
+          map.put(fieldName, row.getValue(fieldName) + "");
+        }
+      } catch (Exception e) {
+        LOG.error(
+            "Error casting "
+                + type.getTypeName().name()
+                + ", "
+                + type.getLogicalType()
+                + ", "
+                + fieldName
+                + ": "
+                + row.getValue(fieldName)
+                + " ["
+                + e.getMessage()
+                + "]");
+        map.put(fieldName, row.getValue(fieldName) + "");
+      }
+    }
+    for (Mapping m : target.mappings) {
+      // if row is empty continue
+      if (listFullOfNulls(row.getValues())) {
+        continue;
+      }
+      String fieldName = m.field;
+      PropertyType targetMappingType = m.type;
+      // lookup data type
+      if (StringUtils.isNotEmpty(m.constant)) {
+        if (StringUtils.isNotEmpty(m.name)) {
+          map.put(m.name, m.constant);
+        } else {
+          map.put(m.constant, m.constant);
+        }
+      }
+    }
+
+    // LOG.info("Casted map: "+mapToString(map));
+    return map;
+  }
+
+  private static boolean listFullOfNulls(List<Object> entries) {
+    for (Object key : entries) {
+      if (key != null) {
+        return false;
+      }
+    }
+    return true;
+  }
+
+  private static String mapToString(Map<String, ?> map) {
+    String mapAsString =
+        map.keySet().stream()
+            .map(key -> key + "=" + map.get(key))
+            .collect(Collectors.joining(", ", "{", "}"));
+    return mapAsString;
+  }
+
+  private static byte[] asBytes(Object obj) throws IOException {
+    ByteArrayOutputStream bytesOut = new ByteArrayOutputStream();
+    ObjectOutputStream oos = new ObjectOutputStream(bytesOut);
+    oos.writeObject(obj);
+    oos.flush();
+    byte[] bytes = bytesOut.toByteArray();
+    bytesOut.close();
+    oos.close();
+    return bytes;
```

Comment:
Prefer to wrap around try-with-resources https://docs.oracle.com/javase/tutorial/essential/exceptions/tryResourceClose.html